### PR TITLE
Optimize episode parser

### LINKED ISSUE #2839: TV-Series episode detection improvement
**Problem:**

TV documentations(mkv) in folders named like this:

Earth.From.Space.2019.E01.WEB-DL.1080p.H264.AAC.2Audio.mkv
Earth.From.Space.2019.E02.WEB-DL.1080p.H264.AAC.2Audio.mkv

were all detected as season 20 episode 19.

**Expect:**

Detect as season 1 episode 01/02

### PR DESCRIPTION
**Pull request by dearjoey**
**Changes**

**Issues**

Fixes #2839

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

File: Emby.Naming/Common/NamingOptions.cs [hunk 1]
```diff
@@ -250,6 +250,8 @@ public NamingOptions()
                 },
                 // <!-- foo.ep01, foo.EP_01 -->
                 new EpisodeExpression(@"[\._ -]()[Ee][Pp]_?([0-9]+)([^\\/]*)$"),
+                // <!-- foo.E01., foo.e01. -->
+                new EpisodeExpression(@"[^\\/]*?()\.?[Ee]([0-9]+)\.([^\\/]*)$"),
```
Comment (cvium) [inline Emby.Naming/Common/NamingOptions.cs:254]: Why does this expression differ from the one above?

Comment (dearjoey) [reply Emby.Naming/Common/NamingOptions.cs:254]: The new one solves the case like
```
The.Legend.of.Condor.Heroes.2017.E07.V2.web-dl.1080p.h264.aac-hdctv.mkv
```
It's a very common name format in PT，this format offen like: xx.xxx.E01.xxxx.xxx
but the old can't support this format

Comment (cvium) [reply Emby.Naming/Common/NamingOptions.cs:254]: I'm asking why your new expression starts with `[^\\/]*?()\.?` when the other one starts with `[\._ -]()`.

Wouldn't changing the expression above to `[\._ -]()[Ee][Pp]?_?([0-9]+)([^\\/]*)$` work just as well (I added `?` after `[Pp]`)?

Comment (dearjoey) [reply Emby.Naming/Common/NamingOptions.cs:254]: If you use your way, this test case will run failed
https://github.com/jellyfin/jellyfin/blob/0c9b64de4be633f2a4f47244996c8a25bbb0db45/tests/Jellyfin.Naming.Tests/TV/EpisodeNumberTests.cs#L49
this case will be recognized as episode 15

Comment (cvium) [reply Emby.Naming/Common/NamingOptions.cs:254]: I'm not sure what `2009x03-E15` is supposed to mean. Season 2009 Episode 3, but what is E15?

Comment (dearjoey) [reply Emby.Naming/Common/NamingOptions.cs:254]: I do not know either what is E15, so I add new one. Do you have a better suggestion?

Comment (Artiume) [reply Emby.Naming/Common/NamingOptions.cs:254]: I would read it as year 2009, season 3, episode 15 or something to that affect

**Review by cvium — APPROVED** [review]

**Pull request merged**